packages/documentsearch/src/searchview.tsx

```diff
@@ -77,13 +78,20 @@ export interface ISearchKeyBindings {
 function SearchInput(props: ISearchInputProps): JSX.Element {
   const [rows, setRows] = useState<number>(1);
 
-  const updateRows = useCallback(
+  const updateDimensions = useCallback(
     (event?: React.SyntheticEvent<HTMLTextAreaElement>) => {
       const element = event
         ? (event.target as HTMLTextAreaElement)
         : props.inputRef?.current;
       if (element) {
-        setRows(element.value.split(/\n/).length);
+        const split = element.value.split(/\n/);
+        // use the longest string out of all lines to compute the width.
+        let longest = split.reduce((a, b) => (a.length > b.length ? a : b), '');
+        if (element.parentNode && element.parentNode instanceof HTMLElement) {
+          element.parentNode.dataset.value = longest;
+        }
+        // element.parentNode?.dataset.value = longest;
```

Comment:
dead code here, will delete along with any other possible changes that need to be made.